crit-go/scripts/magic_gen.go

```diff
@@ -0,0 +1,115 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"github.com/thoas/go-funk"
+	"os"
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+func main() {
+
+	if len(os.Args) != 3 {
+		fmt.Println("Usage: magic-gen.py path/to/image.h path/to/magic.json")
+		os.Exit(1)
+	}
+	magic_c_header := os.Args[1]
+	magic_json := os.Args[2]
+	// opens magic file
+	infile, err := os.Open(magic_c_header)
+	check(err)
+	defer infile.Close()
+	outfile, err := os.OpenFile(magic_json, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	check(err)
+	defer outfile.Close()
+	mapper := make(map[string]interface{})
+	all_magic := make(map[string]string)
+	magic := make(map[string]string)
+	by_name := make(map[string]interface{})
+	by_val := make(map[uint64]interface{})
+	scanner := bufio.NewScanner(infile)
+	scanner.Split(bufio.ScanLines)
+
+	var txtlines []string
+
+	for scanner.Scan() {
+		txtlines = append(txtlines, scanner.Text())
+	}
+
```

Comment:
`scanner.Error()` check is missing here.

Reply:
error check added